tests/testswitch.nim

```diff
@@ -91,18 +91,26 @@ suite "Switch":
 
       var peerInfo1, peerInfo2: PeerInfo
       var switch1, switch2: Switch
-      (switch1, peerInfo1) = createSwitch(ma1)
       var awaiters: seq[Future[void]]
-      awaiters.add(await switch1.start())
 
+      (switch1, peerInfo1) = createSwitch(ma1)
+
+      let testProto = new TestProto
+      testProto.init()
+      testProto.codec = TestCodec
+      switch1.mount(testProto)
       (switch2, peerInfo2) = createSwitch(ma2)
+      awaiters.add(await switch1.start())
       awaiters.add(await switch2.start())
-      var conn = await switch2.dial(switch1.peerInfo)
+      await switch2.connect(switch1.peerInfo)
+      let conn = await switch2.dial(switch1.peerInfo, TestCodec)
```

Comment:
Why call dial() after connect()? Both will call internalConnect().

If it's for testing purposes, add a comment. People usually look at tests for API usage hints.

Reply:
Yeah I think this confused me when doing noise test (also had to add outgoing flag for this reason likely) see https://github.com/status-im/nim-libp2p/pull/90#discussion_r383709965